src/libraries/System.Text.RegularExpressions/tests/Regex.Replace.Tests.cs

```diff
@@ -333,7 +340,7 @@ public void Replace_Invalid()
             if (PlatformDetection.IsNetCore)
             {
                 // Substitutions not supported in DFA mode
-                Assert.Throws<NotSupportedException>(() => new Regex("pattern", RegexHelpers.RegexOptionNonBacktracking).Replace("input", "$0", -1));
+                Assert.Throws<NotSupportedException>(() => new Regex("pattern", RegexHelpers.RegexOptionNonBacktracking).Replace("input", "$1", -1));
```

Comment:
This changed because $0 is now supported?  Are there any tests that validate that?

Reply:
Yes, that's the reason for the change. I'll add a test for $0 directly ($& does parse to the same thing internally, so it does work, but a test will be good to have.)